rotkehlchen/externalapis/beaconchain/service.py

```diff
@@ -360,3 +367,78 @@ def get_eth1_address_validators(self, address: ChecksumEvmAddress) -> list[Valid
             ) from e
         return validators
+
+    def query_validator_daily_stats(
+            self,
+            validator_index: int,
+            last_known_timestamp: Timestamp,
+            exit_ts: Timestamp | None,
+    ) -> list[ValidatorDailyStats]:
+        """
+        May raise:
+        - RemoteError if we can't query beaconcha.in or if the data is not in the expected format
+        """
+        if exit_ts is not None and last_known_timestamp > exit_ts:
+            log.debug(f'Validator {validator_index=} is up to date')
+            return []  # nothing new to add
+
+        data = cast(list[dict[str, Any]], self._query(
+            module='validator',
+            endpoint='stats',
+            encoded_args=str(validator_index),
+        ))
+
+        timestamp, pnl = Timestamp(0), ZERO
+        stats: list[ValidatorDailyStats] = []
+        for day_data in data:
+            try:
+                start_effective_balance = day_data['start_effective_balance']
+                if start_effective_balance is None:
+                    start_effective_balance = ZERO
+                elif deserialize_int(start_effective_balance) == ZERO:
+                    continue  # validator exited
+
+                end_effective_balance = None
+                if day_data['end_effective_balance'] is not None:
+                    end_effective_balance = deserialize_int(day_data['end_effective_balance'])
+
+                try:
+                    timestamp = create_timestamp(
+                        datestr=f"{day_data['day_start']}",
+                        formatstr='%Y-%m-%dT%I:%M:%S%z',
+                    )
+                except ValueError as e:
+                    raise RemoteError(
+                        f'Failed to parse {day_data["day_start"]} to timestamp',
+                    ) from e
+
+                if timestamp <= last_known_timestamp:
+                    return stats  # we are done
+
+                end_balance = start_balance = 0
+                if (raw_end_balance := day_data['end_balance']) is not None:
+                    end_balance = deserialize_int(raw_end_balance)
+
+                if (raw_start_balance := day_data['start_balance']) is not None:
+                    start_balance = deserialize_int(raw_start_balance)
+
+            except KeyError as e:
+                log.error(f'Missing key {e} in daily stats for {validator_index=}. Skipping day')
+                continue
+
+            except DeserializationError as e:
+                log.error(f'Failed to deserialize daily stats for {validator_index=} due to {e}. Skipping day')  # noqa: E501
+                continue
+
+            pnl_as_int = end_balance - start_balance
+            if end_effective_balance == 0:  # on exit ensure that pnl doesn't contain the exited amount  # noqa: E501
+                pnl_as_int -= start_effective_balance
+
+            pnl = FVal(pnl_as_int) / 10 ** 9  # beaconchain uses 9 decimals
```

Comment:
So if this is always in gwei you need to specify it. We have a function for this. `from_gwei`. Use it and this way you can remove the comment and simplify since it also accepts ints!

```suggestion
            pnl = from_gwei(pnl_as_int)
```